### PR TITLE
Client/types sort

### LINKED ISSUE #872: [FEATURE] option to alphabetize types in type list.
**Is your feature request related to a problem? If so, Please describe.**

When there are a ton of features, an alphabetical list is better than sorting by count.

**Describe alternatives you've considered**

At some point in the past, types with no visible tracks used to disappear. What happend there?

### PR DESCRIPTION
**Pull request by subdavis**
Replaces #895 
fixes #872 

Screenshot from 2021-08-19 10-58-23

### CODE REVIEW

**Review by marySalvi — APPROVED** [review]
Looks good.

**Review by BryonLewis — CHANGES_REQUESTED** [review]
Just some questions about the sorting mechanism.

Additionally I did some testing on larger datasets, quickly I couldn't see much of an impact vs the main branch?
Used the dive cli: `dive generate-data --video --frames 120000 --fps 60 --tracks 10000 --types 500` and confidence filtering seems a bit slow when compared with main

**Comment by subdavis** [comment]
I've wasted a lot of time on this.  For whatever reason, during confidence scrubbing, Hammer.js (the ancient touchscreen library) causes the page to freeze because it's using all available cycles.

I've tried to disable it, but webpack configuration `strictModuleExceptionHandling` is not resulting in the correct bundle.  I've found the related code on github, but looking in `node_modules`, the code is totally different for the same tagged version.  This means I cannot force the `hammerjs` global variable to be undefined -- webpack initializes it with the empty object by default when that config isn't set.

## Next steps

* Find some way to force webpack not to define the variable (impossible since webpack seems to be ignoring `strictModuleExceptionHandling`, AFAICT)
* Make a PR to GeoJS to explicitly disable hammer in some kind of constructor args rather than relying on the implicit behavior of "well, if it's possible to `require()`, we can use it"

Don't ask me how this PR relates to hammer performance issues, I have no idea.  

This PR is stalled until the issue can be resolved.

File: client/src/components/TypeList.vue [hunk 1]
```diff
@@ -112,6 +118,34 @@ export default defineComponent({
...
+      const trackType = confidencePair[0];
+      acc.set(trackType, (acc.get(trackType) || 0) + 1);
+
+      return acc;
+    }, new Map<string, number>()));
+
+    function sortTypes(types: Ref<readonly string[]>) {
+      switch (sortingMethods[data.sortingMethod]) {
+        case 'a-z':
+          return types.value.slice(0).sort();
```
[the remark this answers is not in the document]
Comment (subdavis) [reply client/src/components/TypeList.vue:132]: This is a significant issue.  I don't think it's possible to fix, so we either take the performance hit or don't have the feature.

[the remark this answers is not in the document]
Comment (BryonLewis) [reply client/src/components/TypeList.vue:132]: just for clarification are you seeing a significant performance hit when using the new sorting method vs the older counting method?  Or are you talking about the idea of doing the proper alpha/numerical sort?

Another thing may be to change the slider from 'input' update rates to 'on change; update rate.  Or maybe do it after the types get above 100 or something like that?

[the remark this answers is not in the document]
Comment (subdavis) [reply client/src/components/TypeList.vue:132]: The code on this branch adds a performance hit.  Much of it is related to hammer when running the chrome profiler.  I am stumped about it.

I don't think we should change how the slider works.  You need the visual feedback of seeing tracks appear or disappear to use the slider, so without that feedback (and especially if the feedback changes randomly -- users expect it to work a certain way, so changing it based on type count wouldn't be a good choice IMO) users might get confused.

**Pull request closed without merging**